src/PaymenntClient.php

```diff
@@ -51,19 +116,349 @@ public function getCheckout(string $checkoutId) : Checkout {
     return $this->parseCheckoutResult($result);
   }
 
-  /***********************************************************************************************
-  * CHECKOUT API CALLS
-  */
-
   public function createWebCheckout(WebCheckoutRequest $webCheckoutRequest) : Checkout {
     if (!isset($webCheckoutRequest)) {
       throw new Exception("request cannot be null or empty");
     }
+    $webCheckoutRequest->validate();
     $url = PaymenntClient::URL_CHECKOUT_WEB;
     $result = $this->apiCall($url, $webCheckoutRequest);
     return $this->parseCheckoutResult($result);
   }
 
+  public function createMobileCheckout(MobileCheckoutRequest $mobileCheckoutRequest) : Checkout {
+    if (!isset($mobileCheckoutRequest)) {
+      throw new Exception("request cannot be null or empty");
+    }
+    $mobileCheckoutRequest->validate();
+    $url = PaymenntClient::URL_CHECKOUT_MOBILE;
+    $result = $this->apiCall($url, $mobileCheckoutRequest);
+    return $this->parseCheckoutResult($result);
+  }
+
+  public function createQRCheckout(QRCheckoutRequest $qrCheckoutRequest) : Checkout {
+    if (!isset($qrCheckoutRequest)) {
+      throw new Exception("request cannot be null or empty");
+    }
+    $qrCheckoutRequest->validate();
+    $url = PaymenntClient::URL_CHECKOUT_LINK;
+    $result = $this->apiCall($url, $qrCheckoutRequest);
+    return $this->parseCheckoutResult($result);
+  }
+ 
+  public function createLinkCheckout(LinkCheckoutRequest $linkCheckoutRequest) : Checkout {
+    if (!isset($linkCheckoutRequest)) {
+      throw new Exception("request cannot be null or empty");
+    }
+    $linkCheckoutRequest->validate();
+    $url = PaymenntClient::URL_CHECKOUT_LINK;
+    $result = $this->apiCall($url, $linkCheckoutRequest);
+    return $this->parseCheckoutResult($result);
+  }
+
+  public function searchCheckoutRequest(SearchCheckoutRequest $searchCheckoutRequest) : CheckoutPage {
```

Comment:
change the search.. to ...Lookup. i.e. searchCheckoutRequest -> checkoutLookupRequest